application/cli-app/src/main/java/org/mobilitydata/gtfsvalidator/Main.java

```diff
@@ -54,14 +57,30 @@ public static void main(String[] args) {
 
                 filenameList.addAll(config.validateAllOptionalFileName().execute());
 
-                // base validation
+                final List<String> toLoadIntoMemory = new ArrayList<>();
+                // at present this list is hard coded for our needs: agency.txt and routes.txt.
+                toLoadIntoMemory.add("agency.txt");
+                toLoadIntoMemory.add("routes.txt");
+
+                // base validation + load gtfs entities into memory
                 filenameList.forEach(filename -> {
                     config.validateHeadersForFile(filename).execute();
                     config.validateAllRowLengthForFile(filename).execute();
 
                     ParseSingleRowForFile parseSingleRowForFile = config.parseSingleRowForFile(filename);
                     while (parseSingleRowForFile.hasNext()) {
-                        config.validateGtfsTypes().execute(parseSingleRowForFile.execute());
+                        final ParsedEntity parsedEntity = parseSingleRowForFile.execute();
+                        config.validateGtfsTypes().execute(parsedEntity);
+
+                        // load gtfs entities into memory
+                        // in the future all filename in filenameList will be processed. For now focusing on routes.txt
+                        // and agency.txt.
+                        // filenames in filenameList will be determined using a dependency tree defined by a JSON file,
+                        // and command lines or configuration file will be used to exclude files from the validation
+                        // process.
+                        if (toLoadIntoMemory.contains(filename)) {
+                            config.loadAllEntityIntoMemory().execute(parsedEntity);
```

Comment:
Retrieve use case instance before while loop and call execute on it with each parsed entity